src/ArgbColor.h

```diff
@@ -0,0 +1,80 @@
+#ifndef BLACKHOLERAYTRACER_ARGBCOLOR_H
+#define BLACKHOLERAYTRACER_ARGBCOLOR_H
+
+#include "CGL/color.h"
+
+using namespace CGL;
+
+class ArgbColor : public Color{
+public:
+  float a = 1; // alpha channel
+  static const ArgbColor White;
+  static const ArgbColor Black;
+  static const ArgbColor Transparent;
+
+  ArgbColor( float a = 0, float r = 0, float g = 0, float b = 0 )
+        : a(a), Color(r,g,b) { }
+
+  ArgbColor(Color c) : Color(c.r, c.g, c.b) {}
+
+  // Array-style access
+  float& operator[](int k) {
+    return (&r)[k];
```

Comment:
Isn't `r` a float? What does it mean to index into a `float`?

Reply:
Hmm looking at this, for argb it should be `return (&a)[k];` because the format would be in argb order with 8 bits each so 32 bits total.

It's not really indexing into the float but its getting the address of the first channel (which should be alpha channel in this case not r) and its incrementing the address by k * 8 bits to return either the a, r, g, or b. For Color, the fields should be stored next to each other which makes it work but idk how this works for subclassing.

I didn't really look at porting the array stuff from Color to ArgbColor which I made a note of in the ArgbColor.cpp so don't use it. So no indexing into ArgbColor or using the array constructor for ArgbColor.